src/background/main.js

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import {Logger} from "./logger.js";
+import {RequestHandler} from "./requestHandler.js";
+import {TabHandler} from "./tabHandler.js";
+
+const log = Logger.logger("Main");
+
+class Main {
+  constructor() {
+    this.observers = new Set();
```

Comment:
This is ofc totally valid code. 
Given this constructor does not do anything fun, this could just be done in the class description and keep the default constructor, whatever you prefer tho. (Oftopic, should we pick a styleguide? )
```
class Main {
observers = new Set();
logger = new Logger(this);
requestHandlder = new RequestHandler(this);
tabHandler = new TabHandler(this);
.... 
}
```

Reply:
Leaving as is for now since I see you've improved this in #6 (how cute are these single digit PR numbers?)